File: bookstore_backend/src/main/java/com/reins/bookstore/service/CartService.java
package com.reins.bookstore.service;

import com.reins.bookstore.entity.Cart;
import com.reins.bookstore.entity.compositePK.CartPK;
import net.sf.json.JSONObject;

import java.util.List;

public interface CartService {
    Cart addBook2Cart(Integer userId, Integer bookId, Integer addCount);

    List<JSONObject> getCartByUserId(Integer userId);

    Cart deleteBookFromCart(Integer userId, Integer bookId);
}

File: bookstore/src/services/cartService.js
import config from 'config';
import {postRequest, postRequest_v2} from "../utils/ajax";

export const addBook2Cart = (userId, bookId, addCount, callback) => {
    const url = `${config.apiUrl}/addBook2Cart`;
    const data = {userId: userId, bookId: bookId, addCount: addCount};
    postRequest_v2(url, data, callback)
}

export const deleteBookFromCart = (userId, bookId, callback) => {
    const url = `${config.apiUrl}/deleteBookFromCart`;
    const data = {userId: userId, bookId: bookId};
    postRequest_v2(url, data, callback)
}


export const getCartByUserId = (userId, callback) => {
    const url = `${config.apiUrl}/getCartByUserId`;
    const data = {userId: userId};
    postRequest_v2(url, data, callback)
}